FirebaseAuth/Sources/AuthProvider/Phone/FIRPhoneAuthProvider.m

```diff
@@ -593,6 +595,10 @@ - (void)verifyClientAndSendVerificationCodeToPhoneNumber:(NSString *)phoneNumber
  */
 - (void)verifyClientWithUIDelegate:(nullable id<FIRAuthUIDelegate>)UIDelegate
                         completion:(FIRVerifyClientCallback)completion {
+  if ([GULAppEnvironmentUtil isSimulator]) {
+    [self reCAPTCHAFlowWithUIDelegate:UIDelegate completion:completion];
```

Comment:
I'm wondering if there's actually something going wrong in our fallback logic:

As per the reported issue, APNS is failing with INVALID_APP_CREDENTIAL: https://github.com/firebase/firebase-ios-sdk/issues/10214#issue-1373420565. However, we should be falling back to reCAPTCHA automatically in that case: https://github.com/firebase/firebase-ios-sdk/blob/bba19ad013bdd08eae551636e9e345642afb24a9/FirebaseAuth/Sources/AuthProvider/Phone/FIRPhoneAuthProvider.m#L624

That might be worth looking into a little before hard-coding simulators to always use reCAPTCHA.

Reply:
I don't think we want to fallback to recaptcha in that case. If APNs is available but the app credential is invalid, there is a good chance the the app is not legitimate, so we should just error out.